engine/execution/computation/computer/result_collector.go

```diff
@@ -226,6 +231,13 @@ func (collector *resultCollector) commitCollection(
 		NumberOfCollections: 1,
 	}
 
+	for _, consumer := range collector.consumers {
+		err = consumer.OnExecutedCollection(collector.result.CollectionResult(collection.collectionIndex))
```

Comment:
⚠️  `CollectionResult` might return `nil` here, I'm not sure if `OnExecutedCollection` is able to handle `nil` value. 

I would suggest to let `CollectionResult` return `*ColResSnapshot, bool` instead.